packages/ts/lit-form/src/Models.ts

```diff
@@ -213,39 +230,53 @@ export class ArrayModel<MItem extends AbstractModel = AbstractModel> extends Abs
 
   [_createEmptyItemValue]: () => Value<MItem>;
 
-  readonly #createItem: (parent: this, index: number) => MItem;
+  // The `parent` parameter is AbstractModel here for purpose; for some reason, setting it to `ArrayModel<MItem>` or
+  // `this` breaks the type inference in TS (v5.3.2).
+  readonly #createItem: (parent: AbstractModel, index: number) => MItem;
   #items: Array<MItem | undefined> = [];
 
   constructor(
     parent: ModelParent,
     key: keyof any,
     optional: boolean,
-    createItem: (parent: ArrayModel<MItem>, key: number) => MItem,
+    createItem: (parent: AbstractModel, key: number) => MItem,
     options?: ModelOptions<Array<Value<MItem>>>,
   ) {
     super(parent, key, optional, options);
     this.#createItem = createItem;
     this[_createEmptyItemValue] = createItem(this, 0).constructor.createEmptyValue as () => Value<MItem>;
   }
 
-  /**
-   * Iterates the current array value and yields a binder node for every item.
-   */
-  *[Symbol.iterator](): IterableIterator<BinderNode<MItem>> {
-    const array = this.valueOf();
+  *[_items](): Generator<MItem, void, void> {
+    const values = getBinderNode(this).value ?? [];
 
-    if (array.length !== this.#items.length) {
-      this.#items.length = array.length;
+    if (values.length !== this.#items.length) {
+      this.#items.length = values.length;
     }
 
-    for (let i = 0; i < array.length; i++) {
+    for (let i = 0; i < values.length; i++) {
       let item: MItem | undefined = this.#items[i];
 
       if (!item) {
         item = this.#createItem(this, i);
         this.#items[i] = item;
       }
 
+      yield item;
+    }
+  }
+
+  /**
+   * Iterates over the current model and yields a binder node for every item
+   * model.
+   *
+   * @deprecated Use the {@link m.items} function instead. E.g.:
+   * ```ts
+   * const itemValues = Array.from(m.items(model), (itemModel) => m.value(itemModel));
+   * ```
```

Comment:
`m.value` is not available yet :-)

Maybe it's fine to use some use-case specific examples for Lit and React:

```suggestion
   * @deprecated Use the {@link m.items} function instead. For example, in React:
   * ```tsx
   * const {model, field} = useForm(GroupModel);
   * return Array.from(m.items(model.people), (personModel) => 
   *   <TextField label="Full name" {...field(personModel.fullName)} />
   * );
   * ```
   * In Lit:
   * ```ts
   * return html`${repeat(m.items(this.binder.model.people), personModel => html`
   *   <vaadin-text-field label="Full name" ${field(personModel.fullName)}></vaadin-text-field>
   *  `)}
   * `;
```